tests/integration/v3_grpc_test.go

```diff
@@ -1922,7 +1922,45 @@ func TestV3LargeRequests(t *testing.T) {
 					t.Errorf("#%d: range expected no error, got %v", i, err)
 				}
 			}
+		})
+	}
+}
 
+// TestV3AdditionalGRPCOptions ensures that configurable GRPCAdditionalServerOptions works as intended.
+func TestV3AdditionalGRPCOptions(t *testing.T) {
+	integration.BeforeTest(t)
+	tests := []struct {
+		maxRequestBytes uint
+		grpcOpts        []grpc.ServerOption
+		valueSize       int
+		expectError     error
+	}{
+		{8 * 1024 * 1024, nil, 9 * 1024 * 1024, status.Errorf(codes.ResourceExhausted, "grpc: received message larger than max")},
+		{8 * 1024 * 1024, []grpc.ServerOption{grpc.MaxRecvMsgSize(10 * 1024 * 1024)}, 9 * 1024 * 1024, rpctypes.ErrGRPCRequestTooLarge},
+	}
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("#%d", i), func(t *testing.T) {
```

Comment:
Use `name` to be added above as the test case name.

No need to print the `i` in the following code. The golang test lib will print the `name` automatically when fails.